### ISSUE TITLE
failed to install - this package is not published by Synology 

### ISSUE THREAD

**Issue by scargill** [SynoCommunity/spksrc#1439]
Got this message when I tried my first install - Mosquitto.... any ideas?

**Comment by Diaoul** [comment]
Yes, this needs to be explained on the homepage of SynoCommunity.
You need to set the trust level to the second option (Synology and trusted publishers) and reinstall the package.

**Comment by Diaoul** [comment]
As for editing the configuration file, for now it's done through SSH to the NAS using `vi /usr/local/mosquitto/var/mosquitto.conf`. I agree that's not as easy as a GUI but a GUI is hard to develop and you would use it once or twice so that's not a big deal IMO.

**Comment by scargill** [comment]
Got it - just need to figure out how to set up the Mosquitto package now.. turns on - turns off - but can't find any settings! (like passwords etc)

**Comment by scargill** [comment]
Can you possibly give me a little more info … as a Windows user I’m not even sure we can get VI!   I am VPN’ing through to my DISKSTATION…   so I’m effectively on my home network. From Windows I guess I need to access the config file and also create a passwords file. Can you give me an idea how to do that from a Windows laptop?

**Comment by scargill** [comment]
That did not work well. WINSSH gave me access to the mosquito.conf file – I loaded it into windows – updated it  to include passwords – but I was not allowed permission to either upload that or the passwords file back to Windows. Also it occurred to me that in order to use the config file you have to pass the config file to mosquito --- how would you do that?

 

Pete.

**Comment by Diaoul** [comment]
With WinSCP I think you can edit files remotely if memory serves. Maybe you can try that. Make sure you enabled SSH on the NAS and use root user not admin.

**Comment by scargill** [comment]
AH I used admin! Right…

**Comment by scargill** [comment]
Am I doing something daft. I managed to get Mosquitto to work (I’ve run Mosquitto on PCS before).

 

I then made a passwords.txt file – using the mosquito_passwds tool, turned off anonymous, added the name of the passwords file – both in the config file – re-uploaded the config file and uploaded the passwords file – stopped the service on diskstation, started it… and now I cannot log into Mosquitto…..

**Comment by scargill** [comment]
I put the link to the password file – which is in the same location as the config file as.. 

 

password_file /usr/local/mosquitto/var/passwords.txt

 

The program IS taking note of the config file – because I can turn guest access on and can get in – but I don’t think it’s seeing the passwords file… is above right?

**Comment by scargill** [comment]
Oh dear, I’m not a Linux man – that could be fun.

**Comment by Diaoul** [comment]
SSH to your NAS with PuTTY as root:

    chown mosquitto /usr/local/mosquitto/var/passwords.txt

**Comment by scargill** [comment]
Sorted. Thanks.

**Comment by scargill** [comment]
Thanks – I used WINSCP and just made mosquito the owner (right click) – sorted!!! Simple…  this is really good – as I was not too keen on using a PC for Mosquitto – the Diskstation is much the more solid solution!  Thanks a lot for this.

**Comment by sharathsis123** [comment]
Can somebody help me in installing Mosquitto server in linux.

**Comment by scargill** [comment]
You might want to be a LOT more specific. WHICH Linux? Which version of Linux? Mosquitto or Mosquitto with websockets?

**Comment by sharathsis123** [comment]
Thanks for your response .. 
I want to install on redhat linux 6.5 . Only mosquitto server. Thanks

**Comment by sharathsis123** [comment]
Hi Scargill , Do you have any detail spec on installing Mosquitto on linux with websockets and without websockets .. It will be of great help to me if you can provide the spec . Thanks in advance.

**Comment by ajumalp** [comment]
I managed to install and configure Mosquitto MQTT in my Synology NAS server. 
If interested you can read the [Gist here](https://gist.github.com/ajumalp/0ad2517d15c999cfc440cdf3d623fab8)
